lib/lightning_web/live/workflow_live/edit.ex

```diff
@@ -2802,13 +2851,41 @@ defmodule LightningWeb.WorkflowLive.Edit do
       [type, selected] ->
         socket
         |> set_selected_node(type, selected)
-        |> set_mode(if mode in ["expand", "workflow_input"], do: mode, else: nil)
+        |> set_mode(
+          if mode in ["expand", "workflow_input", "history"], do: mode, else: nil
+        )
 
       nil ->
         socket |> unselect_all()
     end
   end
 
+  # version_tag will never be nil here
+  defp handle_run_selection_history(socket, run_id, version_tag, selected_id) do
+    workflow_id = socket.assigns.workflow.id
+
+    %{run_steps: run_steps} =
+      get_run_steps_and_history(workflow_id, run_id)
+
+    snapshot = snapshot_by_version(workflow_id, version_tag)
+
+    # snapshots currently don't have positions
+    # if we don't do this, the patch wouldn't replace the existing position in params
+    updated_snapshot = Map.put(snapshot, :positions, nil)
```

Comment:
Should we add `positions` as a field to the Snapshot schema instead of doing this manual `Map.put`? Doing this manual manipulation feels fragile and could break if the Snapshot structure changes. Having `positions` as an actual field with a default of `nil` would make the data structure more explicit and eliminate the need for this workaround.

Something like:

```elixir
schema "workflow_snapshots" do
  # ... existing fields ...
  field :positions, :map, default: nil
  # ... rest of schema ...
end
```

What do you think?

Reply:
This is cool because we'll obviously have to support node positions in snapshots too. 
with this I run a migration after right?